service_configuration_lib/spark_config.py

```diff
@@ -430,14 +430,16 @@ def _adjust_spark_requested_resources(
         # TODO(gcoll|COREML-2697): Consider cleaning this part of the code up
         # once mesos is not longer around at Yelp.
         if 'spark.executor.instances' not in user_spark_opts:
-
-            executor_instances = int(user_spark_opts.get(
-                'spark.cores.max',
-                str(DEFAULT_MAX_CORES),
-            )) // executor_cores
+            # spark.cores.max provided, calculate based on (max cores // per-executor cores)
+            if 'spark.cores.max' in user_spark_opts:
+                executor_instances = int(user_spark_opts['spark.cores.max']) // executor_cores
+            # spark.executor.instances and spark.cores.max not provided, the executor instances should at least
+            # be equal to DEFAULT_EXECUTOR_INSTANCES.
+            else:
+                executor_instances = max(DEFAULT_MAX_CORES // executor_cores, DEFAULT_EXECUTOR_INSTANCES)
             user_spark_opts['spark.executor.instances'] = str(executor_instances)
 
-            if user_spark_opts['spark.executor.instances'] == str(DEFAULT_EXECUTOR_INSTANCES):
+            if user_spark_opts['spark.executor.instances'] <= str(DEFAULT_EXECUTOR_INSTANCES):
                 log.warning(
                     f'spark.executor.instances not provided. Setting spark.executor.instances as '
```

Comment:
This log message is confusing to me. 
If `spark.executor.instances` is greater than DEFAULT_EXECUTOR_INSTANCES it could be still be possible that it's not provided, right? Example: spark.cores.max = 100 and spark.executor.cores=2, no?

Reply:
Yes, it is still possible but this was added to warn in case we are using the DEFAULT_EXECUTOR_INSTANCES. Should I add it in every case? Also, this warning does not make sense if DRA is enabled but at this point, it is not clear if DRA will be enabled due to mass rollout executor range. How should I proceed?

Reply:
hmm This is getting confusing with lots of if/else conditions. I think eventually we would want all batches to provide either `executor.instances` or `DRA configs for instances`. Anything else should be removed or warned. 

Remove that warning. 

Maybe just revisit the logic in order: 
if `executor.instances` provided use that:
elif `cores.max` is provided, use that with warning:
else use default value; 

Keep it as a separate api to calculate instance config.